chrome/content/test/README.md

```diff
@@ -1,24 +1,145 @@
+```
+   ...
+   <script type="text/javascript" src="./example_tests.js"></script>
+   ...
+```
+
+* Write the tests  -
+```
+// example_tests.js
+describe("Test Suite for Example.js", function() {
+  
+  var Example = g.Example;
+
+  it("Does something cool", function() {
+    Example.doSomethingCool();
+    expect(Example.cool).toBe("done");
+  });
+
+});
 ```
 
-However unit-tests for `privly-jetpack` are run in a different manner.
+That's it! This should get you started with writing tests for your module.
 
-### Jetpack tests ###
+### How to get a module covered ###
 ---
 
-If we want to test the same `posting_process.js` file, our test_loader.html wouldn't include the posting process script as a script in test_loader.html but would rather be loaded into a object because it's a CommonJS module. Now the object would have access to all the functions in the module because the module exports it.
+(For a better understanding of what's really going on here, read the [Architecture](https://github.com/privly/privly-jetpack/tree/master/chrome/content/test#architecture)/[Code Coverage](https://github.com/privly/privly-jetpack/tree/master/chrome/content/test#code-coverage) section first).
+
+If you want a module(say `lib/example.js`) to be covered for code coverage, you would have to do the following --
+
+* Add a `coverage` function in the module -
+```
+  ...
+  /**
+   * Returns the coverage data for this file/module when tested.
+   */
+  coverage: function() {
+    var cv = require("./coverage_var.js").coverageVar;
+    // You just need to pass the filename.
+    return eval(cv.generate("example.js"));
+  },
+  ...
+```
+
+* Make sure that the XPCOM service function returns a `.in.js` version of the file.
+```
+  ...
+  // part of PrivlyTestXPCOMService
+  getExampleObject: function() {
+    // .in.js is the instrumented version of example.js
+    // the instrumented version is generated using "istanbul instrument" by run_test.sh
+    return ("./example.in.js");
+  },
+  ...
+```
+
+### Architecture ###
+---
+
+`jpm run` launches an instance of Firefox with the extension installed. In order to open the `test_loader.html` page, we pass a `--prefs test.json` argument to the `jpm run` command. `test.json` contains key-value pairs that signals the extension to open up the test loader page. These key-value pairs in `test.json` are stored in the preferences service of the launched Firefox browser.
+
+The test loader page is an HTML page that loads the jasmine library scripts and test scripts. The page is loaded in a `chrome://privly/` URL and the scripts are run in a privileged environment. `globals.js` loads the extension commonjs modules into objects that are used by the test scripts. The extension modules are made available to `globals.js` via an XPCOM service defined in [lib/xpcom.js](https://github.com/privly/privly-jetpack/blob/master/lib/xpcom.js). The test scripts have access to the extension module objects via a global `g` namespace.
```

Comment:
I think the second paragraph is difficult to follow.  Reorganizing this so that definitions of the components and the way the components are connected are presented separately would go a long way in clarifying the contents.